src/Engine/Manager/SoundManager.cpp

```diff
@@ -74,39 +71,91 @@ void SoundManager::CheckError(PaError err) {
 
 void SoundManager::Update(float deltaTime) {
 
+    std::vector<Component::Listener*> listeners = GetListeners();
+    if (listeners.size() == 0)
+        return;
+
     // Number of samples to process dependant on deltaTime
-    int numSamples = int(SAMPLE_RATE * deltaTime);
+    unsigned int frameSamples = int(SAMPLE_RATE * deltaTime);
+    if (frameSamples > CHUNK_SIZE) {
+        Log() << "SoundManager::Update: Frame drop!\n";
+        frameSamples = CHUNK_SIZE;
+    }
+    targetSample += frameSamples;
+
+    // Update sound.
+    while (currentSample < targetSample) {
+        // Process Samples.
+        if (processedSamples == 0) {
+            ProcessSamples();
+            processedSamples = CHUNK_SIZE;
+        }
+
+        // Play samples.
+        unsigned int sampleCount = std::min(targetSample - currentSample, processedSamples);
+        unsigned int index = (CHUNK_SIZE - processedSamples) * 2;
+        Pa_WriteStream(stream, &processedBuffer[index], sampleCount);
+        processedSamples -= sampleCount;
+        currentSample += sampleCount;
+    }
+}
+
+void SoundManager::ProcessSamples() {
+
+    std::vector<Component::Listener*> listeners = GetListeners();
+    if (listeners.size() == 0)
+        return;
 
+    Entity* player = listeners[0]->entity;
+    glm::vec3 glmPos = player->GetWorldPosition();
+    glm::quat orientation = player->GetWorldOrientation();
+    glm::vec3 glmDir = orientation * glm::vec3(0, 0, -1);
+    glm::vec3 glmUp = orientation * glm::vec3(0, 1, 0);
+    IPLVector3 pos = { glmPos.x, glmPos.y, glmPos.z };
+    IPLVector3 dir = { glmDir.x, glmDir.y, glmDir.z };
+    IPLVector3 up = { glmUp.x, glmUp.y, glmUp.z };
+
+    // Set player transform
+    sAudio.SetPlayer(pos, dir, up);
+
+    std::size_t size = soundSources.GetAll().size();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Variable 'size' is assigned a value that is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)